packages/lib/server/repository/watchlist.repository.ts

```diff
@@ -55,6 +45,21 @@ export class WatchlistRepository implements IWatchlistRepository {
     return watchlist;
   }
 
+  async createEntryIfNotExists(params: CreateWatchlistInput): Promise<WatchlistEntry> {
+    const existing = await this.checkExists({
+      type: params.type,
+      value: params.value,
+      organizationId: params.organizationId,
+      isGlobal: params.isGlobal,
+    });
+
+    if (existing) {
+      throw new Error("Watchlist entry already exists for this organization");
+    }
+
+    return this.createEntry(params);
+  }
```

Comment:
When I meant to rename to createEntryIfNotExists, I mean that such a fn shoudn't throw error even if record already exists, i.e. it should be idempotent.  
  
This is similar to `IF NOT EXISTS` clause with create in sql